cou/apps/base.py

```diff
@@ -713,3 +717,40 @@ def _get_wait_step(self) -> PostUpgradeStep:
             parallel=False,
             coro=self.model.wait_for_active_idle(self.wait_timeout, apps=apps),
         )
+
+    def _check_auto_restarts(self) -> None:
+        """Check if enable-auto-restarts is enabled.
+
+        If the enable-auto-restart is not enabled, this check will raise an exception.
+
+        :raises HaltUpgradePlanGeneration: When enable-auto-restarts is not enabled.
+        """
+        if "enable-auto-restarts" not in self.config:
+            logger.debug("%s application has not enable-auto-restarts config option", self.name)
+            return
+
+        if self.config["enable-auto-restarts"].get("value") is False:
+            raise HaltUpgradePlanGeneration(
+                f"It is not safe to continue upgrading '{self.name}' application when the "
+                "'enable-auto-restart' is disabled. Please enable it before running COU again."
```

Comment:
I suggest rewording as follows:
```suggestion
                f"COU does not currently support upgrading applications that disable service restarts. "
                "Please enable charm option enable-auto-restart and rerun COU to upgrade the "
                "{self.name} application."
```